spec/converters/csv_spec.rb

```diff
@@ -0,0 +1,152 @@
+require 'spec_helper'
+require 'ronin/masscan/converters/csv'
+require 'tempfile'
+
+RSpec.describe Ronin::Masscan::Converters::CSV do
+  let(:fixtures_path) { File.expand_path(File.join(__dir__, '..', 'fixtures')) }
+  let(:masscan_path)  { File.join(fixtures_path, 'converter', 'masscan.json') }
+  let(:masscan_file)  { Masscan::OutputFile.new(masscan_path) }
+  let(:ip_addr)       { IPAddr.new('93.184.216.34/32') }
+  let(:timestamp)     { 1629960621 }
+  let(:header)        { Ronin::Masscan::Converters::CSV::HEADER }
+
+  describe '.convert' do
+    let(:expected_row) { "status,open,tcp,80,\"syn,ack\",54,93.184.216.34,#{Time.at(timestamp)}" }
+
+    it 'must convert masscan file to csv and write it into output' do
+      Tempfile.create do |output_file|
+        subject.convert(masscan_file, output_file)
+        output_file.rewind
+
+        expect(output_file.gets.chomp).to eq(header.join(','))
```

Comment:
I'd call `.gets` once and save it to a local variable. Also you can use `.gets(chomp: true)`.

Reply:
Didn't `.gets` gets only one line? I think we should test against first 3 lines.

Reply:
I used `.readlines(chomp: true)` to save it into variable.